tests/baselines/reference/inferTypes2.types

```diff
@@ -32,7 +32,7 @@ export type BadNested<T> = { x: T extends number ? T : string };
 
 export declare function foo2<T>(obj: T): T extends { [K in keyof BadNested<infer P>]: BadNested<infer P>[K] } ? P : never;
 >foo2 : <T>(obj: T) => T extends { x: infer P extends number ? infer P : string; } ? P : never
->     : ^ ^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^
+>     : ^ ^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

Comment:
Hm, how did this end up with _less_ reuse?

Reply:
This one was fixed in the last commit via https://github.com/microsoft/TypeScript/pull/58063#discussion_r1550397124; resolving.